### ISSUE TITLE
Passing read preference

### ISSUE THREAD

**Issue by thekid** [xp-forge/mongodb#3]
https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#passing-read-preference-to-mongos-and-load-balancers

**Comment by thekid** [comment]
Don't send read preference for non-mongos when using OP_QUERY or for all standalones:

```go
if serverKind == description.Standalone || (isOpQuery && serverKind != description.Mongos) || op.Type == Write {
  // Don't send read preference for non-mongos when using OP_QUERY or for all standalones
  return nil, nil
}
```

*(see https://github.com/mongodb/mongo-go-driver/blob/83fefd1b968009442a78a1be273f719f9a14b10e/x/mongo/driver/operation.go#L1188)*

```go
desc.Kind = Standalone

if isReplicaSet {              // from "isreplicaset"
  desc.Kind = RSGhost
} else if desc.SetName != "" { // from "setName"
  if isWritablePrimary {       // from "isWritablePrimary" ?? "ismaster"
    desc.Kind = RSPrimary
  } else if hidden {           // from "hidden"
    desc.Kind = RSMember
  } else if secondary {        // from "secondary"
    desc.Kind = RSSecondary
  } else if arbiterOnly {      // from "arbiterOnly"
    desc.Kind = RSArbiter
  } else {
    desc.Kind = RSMember
  }
} else if msg == "isdbgrid" {  // from "msg"
  desc.Kind = Mongos
}
```

*(see https://github.com/mongodb/mongo-go-driver/blob/83fefd1b968009442a78a1be273f719f9a14b10e/mongo/description/server.go#L294 and https://github.com/mongodb/mongo-go-driver/blob/83fefd1b968009442a78a1be273f719f9a14b10e/mongo/description/server_kind.go#L13)*

**Comment by thekid** [comment]
Implemented in xp-forge/mongodb@20f5a267d5b5397cc6310f6ec7c018579478f04b
